tests/test_decoding.py

```diff
@@ -64,56 +65,52 @@ def objcodes(tnef):
 
 def test_decode(tnefspec):
     fn, key, attchs, body, objs = tnefspec
-    with open(datadir + os.sep + fn, "rb") as tfile:
-        t = TNEF(tfile.read())
-        assert t.key == key, "wrong key: 0x%2.2x" % t.key
+    t = TNEF((datadir / fn).read_bytes())
+    assert t.key == key, f"wrong key: {t.key:#04x}"
 
-        for m in t.mapiprops:
-            assert m.__str__()
-            assert m.data is not None
+    for m in t.mapiprops:
+        assert m.__str__()
+        assert m.data is not None
 
-        for i, a in enumerate(t.attachments):
-            assert a.long_filename() == attchs[i]
-            for m in a.mapi_attrs:
-                assert m.__str__()
-                assert m.data is not None
-
-        for m in t.msgprops:
+    for i, a in enumerate(t.attachments):
+        assert a.long_filename() == attchs[i]
+        for m in a.mapi_attrs:
             assert m.__str__()
             assert m.data is not None
-            if m.name == TNEF.ATTRECIPTABLE:
-                for n_m in m.data[0]:
-                    assert isinstance(n_m, TNEFMAPI_Attribute)
 
-        if t.htmlbody:
-            assert 'html' in t.htmlbody
+    for m in t.msgprops:
+        assert m.__str__()
+        assert m.data is not None
+        if m.name == TNEF.ATTRECIPTABLE:
+            for n_m in m.data[0]:
+                assert isinstance(n_m, TNEFMAPI_Attribute)
+
+    if t.htmlbody:
+        assert 'html' in t.htmlbody
 
-        if body:
-            assert getattr(t, body)
-            assert t.has_body()
-        else:
-            assert not t.has_body()
+    if body:
+        assert getattr(t, body)
+        assert t.has_body()
+    else:
+        assert not t.has_body()
 
-        if t.rtfbody:
-            assert t.rtfbody[0:5] == b'{\\rtf'
+    if t.rtfbody:
+        assert t.rtfbody[0:5] == b'{\\rtf'
 
-        if objs:
-            assert objcodes(t) == objs, "wrong objs: %s" % ["0x%2.2x" % o.name for o in t.objects]
+    if objs:
+        assert objcodes(t) == objs, f"wrong objs: {[f'{o.name:#04x}' for o in t.objects]}"
 
-        assert t.dump(True)
-        assert t.dump(False)
+    assert t.dump(True)
+    assert t.dump(False)
 
 
 def test_zip():
-    with open(datadir + os.sep + 'one-file.tnef', "rb") as tfile:
-        zip_data = to_zip(tfile.read())
-        with tempfile.TemporaryFile() as out:
-            out.write(zip_data)
+    zip_data = to_zip((datadir / "one-file.tnef").read_bytes())
+    with tempfile.TemporaryFile() as out:
+        out.write(zip_data)
 
```

Comment:
I just noticed there is no assert statement. While it missed before, maybe you can add an assert statement? ie peek into the zip file and assert the contained file is present?

I used something like this in my other project's test suite:
```
    zip_stream = io.BytesIO(byte..)
    zip_file = zipfile.ZipFile(zip_stream)
    assert zip_file.namelist() == ['carm.lic']
```

Reply:
Thank you for the hint.